pkg/services/sqlstore/alert.go

```diff
@@ -339,3 +342,47 @@ func GetAlertStatesForDashboard(query *m.GetAlertStatesForDashboardQuery) error
 
 	return err
 }
+
+func GetMissingAlerts(query *m.GetMissingAlertsQuery) error {
+
+	//Get current timestamp
+	var ts int64 = -1
+	err := inTransaction(func(sess *xorm.Session) error {
+		results, err := sess.Query("select " + dialect.CurrentTimeToRoundMinSql() + " as ts ")
+		if err != nil {
+			sqlog.Error("Failed to get timestamp", "error", err)
+			return err
+		}
+		ts, err = strconv.ParseInt(string(results[0]["ts"]), 10, 64)
+		if err != nil {
+			sqlog.Error("Failed to get timestamp", "error", err)
+			return err
+		}
+		return nil
+	})
+	if err != nil {
+		sqlog.Error("Transaction failed", "error", err)
+		return err
+	}
+
+	//Get all alerts
+	cmd := &m.GetAllAlertsQuery{}
+	if err := bus.Dispatch(cmd); err != nil {
+		sqlog.Error("Could not load alerts", "error", err)
+		return err
+	}
+
+	//Find Missing alerts
+	var evalTime int64
+	var frequency int64
+	missedAlerts := make([]*m.Alert, 0)
+	for _, alert := range cmd.Result {
+		evalTime = alert.EvalDate.Unix()
+		frequency = alert.Frequency
+		if evalTime < ts-frequency && evalTime >= ts-2*frequency {
```

Comment:
The criteria seem generally ok but consider having some configuration around how far we want to go back (just as a guard-rail). 

Few of the unlikely scenarios -  
1. frequency could be hourly/daily (incorrectly?) does it make sense to go that far
2. there can be weird cases where for some reason, a run might take more than few (3?) minutes on all nodes (assuming freq =1 min)